src/services/communicationLog.ts

```diff
@@ -0,0 +1,49 @@
+import { WhereOptions, Op } from 'sequelize';
+import db from '../models';
+
+const { CommunicationLog } = db;
+
+const createLog = async (recipientId: number, userId: number, data: unknown) => {
+  const log = await CommunicationLog.create({
+    recipientId,
+    userId,
+    data,
+  });
+  return log;
+};
+
+const logById = async (id: number) => {
+  const log = await CommunicationLog.findByPk(id);
+  return log;
+};
+
+const logsByRecipientAndScopes = async (recipientId: number, scopes: WhereOptions[] = []) => {
+  const logs = await CommunicationLog.findAll({
+    where: {
+      recipientId,
+      [Op.and]: [
+        ...scopes,
+      ],
+    },
+  });
+  return logs;
+};
```

Comment:
just a question, why not just call the findAll directly instead of storing it to a var first?

Reply:
mild brain damage